open-craft/opencraft
# Devan/upgrade django

This PR contains the changes required to upgrade the from the current version of Django. 1.10.7 to 1.11. The main steps taken to achieve this were as follows:

    update SwampDragon to fork that supports Django 1.11
    update Django to 1.11
    resolve django-angular issues

**Dependencies**: None

**Testing instructions**: 
- install and setup opencraft devstack on your local machine (https://github.com/open-craft/opencraft)
- run `make test.unit` to verify installation works
- switch to devan/upgrade-django branch
- run `pip install -r requirements.txt`
- run `make test.unit` to verify installation
- run`make run.dev`
- go to http://localhost:5000/registration/ and register

**Author notes**:

SwampDragon fork that supports Django 1.11:
https://github.com/lepri/swampdragon/commits/master

Django 1.11 removes ChoiceFieldRenderer class:
https://github.com/jrief/django-angular/issues/284

Django 1.11 support for django-angular:
https://github.com/jrief/django-angular/pull/292

Pylint issues with Django 1.11:
https://github.com/PyCQA/pylint/issues/1653

**Reviewers**
